File: 0606-construct-string-from-binary-tree/0606-construct-string-from-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def tree2str(self, root: Optional[TreeNode]) -> str:
        if not root:
            return ""
        
        res = str(root.val)
        
        l = self.tree2str(root.left)
        r = self.tree2str(root.right)
        
        if l:
            res += f"({l})"
        elif r:
            res += "()"
        
        if r:
            res += f"({r})"
        
        return res

File: 0606-construct-string-from-binary-tree/0606-construct-string-from-binary-tree.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return String
     */
    function tree2str($root) {
        if($root === null) {
            return "";
        }
        $res = strval($root->val);
        if($root->right !== null) {
            $res .= "(".$this->tree2str($root->left).")";
            $res .= "(".$this->tree2str($root->right).")";
        }
        else if($root->left !== null) {
            $res .= "(".$this->tree2str($root->left).")";
        }
        return $res;
    }
}